File: setpagesdialog.cpp
#include "setpagesdialog.h"
#include "ui_setpagesdialog.h"

SetPagesDialog::SetPagesDialog(int &columns, int &rows, QWidget *parent) :
    QDialog(parent),
    ui(new Ui::SetPagesDialog)
{
    ui->setupUi(this);
    this->columns=&columns;
    this->rows=&rows;
}

SetPagesDialog::~SetPagesDialog()
{
    delete ui;
}

void SetPagesDialog::on_pushButton_clicked()
{
    *columns=ui->spbColumns->value();
    *rows=ui->spbRows->value();
    this->accept();
}

File: setpagesdialog.py
from ui_setpagesdialog import Ui_SetPagesDialog
from PyQt5.QtWidgets import QDialog

class SetPagesDialog(QDialog, Ui_SetPagesDialog):
    
    def __init__(self, parent=None):
        super().__init__(parent)
        self.setupUi(self)
    
    def on_pushButton_clicked(self):
        self.accept()
